File: 3174-minimum-number-of-changes-to-make-binary-string-beautiful/3174-minimum-number-of-changes-to-make-binary-string-beautiful.py
class Solution:
    def minChanges(self, s: str) -> int:
        res = ''
        count = 0
        for i in range(len(s)):
            res = res + s[i]
            if len(res) == 2 and res[0] != res[1]:
                res = ''
                count+=1
            elif len(res) == 2:
                res = ''
        return count

File: 3174-minimum-number-of-changes-to-make-binary-string-beautiful/3174-minimum-number-of-changes-to-make-binary-string-beautiful.js
/**
 * @param {string} s
 * @return {number}
 */
var minChanges = function (s) {
    let str = ''
    let count = 0
    for (let i = 0; i < s.length; i++) {
        str = str + s[i]
        if (str.length == 2 && str[0] != str[1]) {
            str[0] = str[1]
            count++
            str = ''
        }
        else if (str.length == 2) {
            str = ''
        }
    }
    return count
};